jujugui/static/gui/src/app/init/component-renderers-mixin.js

```diff
@@ -0,0 +1,1080 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const yui = window.yui;
+
+const ComponentRenderersMixin = (superclass) => class extends superclass {
```

Comment:
I agree that this can be done in a follow up, but I am not sure my suggestion is out of scope. The mixin has been introduced in this branch, and I am suggesting an alternative approach using delegation. Given how big this change is I am just worried that these suggestions will be lost in desolate land of old github reviews ;-)